### PR TITLE
fix: kms key does not propapate in register model step

### PR DESCRIPTION
**Pull request by EthanShouhanCheng**
*Issue #, if available:*
https://sim.amazon.com/issues/AML-126368
Customer specified KMS is not propagated to register step; the model is encrypted by managed key instead of the key provided.

*Description of changes:*
Propagate KMS to register step when calling register.

*Testing done:*
Test done with dev account; output of register step is encrypted wit the provided KMS key.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/estimator.py [hunk 1]
```diff
@@ -1007,6 +1007,7 @@ def register(
         if compile_model_family is not None:
             model = self._compiled_models[compile_model_family]
         else:
+            kwargs["model_kms_key"] = self.output_kms_key
```
Comment (danabens) [inline src/sagemaker/estimator.py:1010]: what if `model_kms_key` is already in kwargs?

Comment (EthanShouhanCheng) [reply src/sagemaker/estimator.py:1010]: good catch, I will fix it

**Review by rohangujarathi — APPROVED** [review]

**Review by jerrypeng7773 — APPROVED** [review]

**Review by icywang86rui — COMMENTED** [review]
Do we need to use the specified kms key to encrypt the repacked model tar ball in this script as well - https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/workflow/_repack_model.py

**Comment by EthanShouhanCheng** [comment]
> specified

I think the tar file is encrypted; at lease from what I tested, the .tar file output from the register step is encrypted with the kms key provided. Not sure if I missed anything

**Comment by icywang86rui** [comment]
> > Do we need to use the specified kms key to encrypt the repacked model tar ball in this script as well - https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/workflow/_repack_model.py
> 
> I think the tar file is encrypted; at least from what I tested, the .tar file output from the register step is encrypted with the kms key provided. Not sure if I missed anything

Ohh. I think i know what's going on now. The script only bundles the inference script with the model artifact and put that generated tar ball in /opt/ml/model. This script is running in a training job, the training job is taking care of the part uploading this model to s3 and use that specified kms_key for server side encryption. so as long as the correct kms_key get passed in that SKLearn estimator the output model archive will be encrypted with the right key.

**Review by icywang86rui — APPROVED** [review]

**Pull request merged**